pkg/labels/labels.go

```diff
@@ -81,6 +81,27 @@ func (ls *Labels) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// MatchLabels returns a subset of Labels that matches/does not match with the provided label names based on the 'on' boolean.
+// If on is set to true, it returns the subset of labels that match with the provided label names and its inverse when 'on' is set to false.
+func (ls Labels) MatchLabels(on bool, names ...string) Labels {
+	matchedLabels := make(Labels, 0)
```

Comment:
Nit: We try to avoid `make` for stylistic reasons wherever it's not actually necessary (such as zero-length allocations). So in this case, `var matchedLabels Labels` or `matchedLabels := Labels{}` would be closer to the rest of the codebase.

Reply:
Done!